Topic 3/src/test/java/bootcamp/exercises/FileListTest.java

```diff
@@ -0,0 +1,36 @@
+package bootcamp.exercises;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class FileListTest {
+
+	
+	FileList fileList= new FileList(3);
+	
+	@Test
+	public void test() {
+		assertEquals(fileList.getFiles(),"[]");
+	}
+	
+	@Test
+	public void test2() {
+		fileList.addFile("hello");
+		fileList.addFile("I");
+		fileList.addFile("am");
+		assertEquals(fileList.getFiles(),"[hello, I, am]");
+	}
+	
+	@Test
+	public void test3() {
+		fileList.delete();
+		fileList.setSize(4);
+		fileList.addFile("five");
+		fileList.addFile("ten");
+		fileList.addFile("five");
+		fileList.addFile("nine");
+		assertEquals(fileList.getFiles(),"[ten, five, nine]");
+	}
+
```

Comment:
[High] When you need to create or prepare all tests in the same way you should use @Before. Then, when you need to clean your objects after tests, you should use @After.

*When writing tests, it is common to find that several tests need similar objects created before they can run. Annotating a public void method with @Before causes that method to be run before the Test method. The @Before methods of superclasses will be run before those of the current class*

*If you allocate external resources in a Before method you need to release them after the test runs. Annotating a public void method with @After causes that method to be run after the Test method. All @After methods are guaranteed to run even if a Before or Test method throws an exception. The @After methods declared in superclasses will be run after those of the current class*

Examples:
```java
 public class Example {
    List<String> empty;
    
    @Before
    public void initialize() {
       empty= new ArrayList();
    }
    
    @Test
    public void size() {
       empty.add("Hi");
    }

    @After
     public void tearDown() {
       empty.clear();
     }

```